p2pspec

```diff
@@ -1,4 +1,36 @@
-:::NEW p2p spec:::
+:::POTATO MESSAGES:::
+
+:establishing connection:
+- A->B connect request [conn_uuid, challenge, identity {A, B}]
+  - spam when initiating connection
+  - retry every 10 minutes
+- B->A connect response [conn_uuid, B signed challenge, challenge]
+  - B now stores {conn_uuid, A} in active connections
+- A->B connect response response [conn_uuid, A signed challenge]
+  - A now stores {conn_uuid, B} in active connections
+- A<->B ping
+  - periodic message to indicate the connection is still alive
+
+:blocks:
+- A->B request block
+  - B server caches request {A, blockrange} and starts serving
+  - A should spam this request to ensure receipt
+    - can stop spamming when first block is received
+- B->A send block (either in response to request or when new block is made)
+  - how should A re-request dropped blocks?
+
+:potato TCP msg:
+- TCP nonsense built over UDP
+  - prob easier just to use TCP...
+
```

Comment:
i was going to rewrite it a bit to distinguish verifier to verifier messages and player to verifier messages